src/twister2/plugin.py

```diff
@@ -317,3 +329,18 @@ def pytest_runtest_makereport(item, call):
     if report.failed:
         setattr(item, '_test_failed', True)
     return report
+
+
+def pytest_collection_modifyitems(
```

Comment:
I wonder if there shouldn't be added `@pytest.hookimpl(hookwrapper=True, trylast=True)` decorator? Unfortunately `pytest_collection_modifyitems` is used in three more places in all Twister plugin, for example there:
https://github.com/zephyrproject-rtos/twister/blob/b9df9d2bbd4c8feb5458f223f4eb156b04d102fb/src/twister2/generate_tests_plugin.py#L115-L131
or there:
https://github.com/zephyrproject-rtos/twister/blob/b9df9d2bbd4c8feb5458f223f4eb156b04d102fb/src/twister2/report/test_plan_plugin.py#L64-L81

So maybe to avoid problem with hook calling order, this one should be marked explicitly.